Remotion/Core/UnitTests/Context/SafeContextTests/SafeContextTest.ExecutionContext.cs

```diff
@@ -0,0 +1,58 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Threading;
+using NUnit.Framework;
+using Remotion.Context;
+
+namespace Remotion.UnitTests.Context.SafeContextTests
+{
+  [TestFixture]
+  public class SafeContextExecutionContextTest : SafeContextTestBase
+  {
+    [Test]
+    public void Run_WithImplicitSafeContextStorageProvider_OpensSafeContextBoundary ()
+    {
+      var safeContextStorageProvider = _safeContextProviderMock.Object;
+      using (SetupImplicitSafeContextStorageProvider(safeContextStorageProvider))
+        RunThreadStartTest((callback, state) => SafeContext.ExecutionContext.Run(ExecutionContext.Capture()!, callback, state));
```

Comment:
unit tests don't have NRT enabled so why the null-forgiving?

Reply:
ReSharper/Rider still emits a warning for a possible null assignment. Do you want to remove it anyways?

Reply:
hmm... open a discussion on this in MSTeams?